File: frontend/src/app/models/Produit.ts
import {Avis} from "./Avis";
import {Categorie} from "./Categorie";
import {Variation} from "./Variation";


export class Produit {

  id!: number;
  nom!: string;
  description!: string;
  image!: string;
  quantite!:number;
  prix!:number;
  categorie!: Categorie;

  variations!: Variation[];
}

File: backend/src/main/java/com/nigelle/ecommerceback/entities/Produit.java
package com.nigelle.ecommerceback.entities;


import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Produit {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;

    private String description;
    private String image;
    private int quantite;
    private double prix;

    @ManyToOne
    private Categorie categorie;

}
